kamu-cli/src/explore/api_server.rs

```diff
@@ -27,6 +35,14 @@ impl APIServer {
                 "/graphql",
                 axum::routing::get(graphql_playground).post(graphql_handler),
             )
+            .nest(
+                format!("/:{}", kamu_adapter_http::PARAMETER_DATASET_NAME).as_str(),
+                kamu_adapter_http::create_dataset_transfer_protocol_routes()
+                    .layer(axum::middleware::from_fn(
+                        kamu_adapter_http::resolve_dataset_by_name,
+                    ))
+                    .layer(axum::extract::Extension(local_dataset_resolver)),
+            )
```

Comment:
This is great - almost how I imagined it. 

I wonder if we can further simplify it to something like this:
```rust
.nest(
    "/:datasetName",
    kamu_adapter_http::create_dataset_transfer_protocol_routes()
        .layer(kamu_adapter_http::LocalDatasetResolver(
            |request| odf::DatasetName::try_parse(request.path_params.get("datasetName"))?.as_local_ref()),
        ),
)
````
i.e. have just one `DatasetResolverLocal` middleware that we can parametrize with a lambda that knows how to extract a `DatasetRefLocal` from request parts.

Reply:
Solved in your branch